code/components/citizen-server-impl/src/state/ServerGameState.cpp

```diff
@@ -4538,6 +4539,169 @@ struct CStartProjectileEvent
     MSGPACK_DEFINE_MAP(ownerId, projectileHash, weaponHash, initialPositionX, initialPositionY, initialPositionZ, targetEntity, firePositionX, firePositionY, firePositionZ, effectGroup, unk3, commandFireSingleBullet, unk4, unk5, unk6, unk7, unkX8, unkY8, unkZ8, unk9, unk10, unk11, throwTaskSequence, unk12, unk13, unk14, unk15, unk16);
 };
 
+/*NETEV ptFxEvent SERVER
+/#*
+ * Triggered when a particle fx (ptFx) is created.
+ *
+ * @param sender - The ID of the player that triggered the event.
+ * @param data - The event data.
+ #/
+declare function ptFxEvent(sender: number, data: {
+	effectHash: number,
+	assetHash: number,
+	posX: number,
+	posY: number,
+	posZ: number,
+	offsetX: number,
+	offsetY: number,
+	offsetZ: number,
+	rotX: number,
+	rotY: number,
+	rotZ: number,
+	scale: number,
+	axisBitset: number,
+	isOnEntity: boolean,
+	entityNetId: number,
+	f109: boolean,
+	f92: number,
+	f110: boolean,
+	f105: number,
+	f106: number,
+	f107: number,
+	f111: boolean,
+	f100: number
+}): void;
+*/
+struct CNetworkPtFXEvent
+{
+	void Parse(rl::MessageBuffer& buffer)
+	{
+		effectHash = buffer.Read<uint32_t>(32);
+		assetHash = buffer.Read<uint32_t>(32);
+
+		int _posX = buffer.ReadSignedFloat(19, 27648.0f);
+		int _posY = buffer.ReadSignedFloat(19, 27648.0f);
+		int _posZ = buffer.ReadFloat(19, 4416.0f) - 1700.0f;
+
+		rotX = buffer.ReadSignedFloat(19, 27648.0f) * toDegrees;
+		rotY = buffer.ReadSignedFloat(19, 27648.0f) * toDegrees;
+		rotZ = (buffer.ReadFloat(19, 4416.0f) - 1700.0f) * toDegrees;
+
+		scale = (buffer.Read<int>(10) / 1023.0f) * 10.0f;
+
+		axisBitset = buffer.Read<uint8_t>(3);
+
+		isOnEntity = buffer.Read<uint8_t>(1);
+
+		if (isOnEntity)
+		{
+			posX = 0.0f;
+			posY = 0.0f;
+			posZ = 0.0f;
+
+			offsetX = _posX;
+			offsetY = _posY;
+			offsetZ = _posZ;
+
+			entityNetId = buffer.Read<uint16_t>(13);
+		}
+		else
+		{
+			posX = _posX;
+			posY = _posY;
+			posZ = _posZ;
+
+			offsetX = 0.0f;
+			offsetY = 0.0f;
+			offsetZ = 0.0f;
+
+			entityNetId = 0;
+		}
+
+		f109 = buffer.Read<uint8_t>(1);
+
+		if (f109)
+		{
+			f92 = buffer.Read<int>(32);
+		}
+		else
+		{
+			f92 = -1;
+		}
+
+		f110 = buffer.Read<uint8_t>(1);
+
+		if (f110)
+		{
+			f105 = buffer.Read<uint16_t>(8);
+			f106 = buffer.Read<uint16_t>(8);
+			f107 = buffer.Read<uint16_t>(8);
+		}
+		else
+		{
+			f105 = 0;
+			f107 = 0;
+		}
+
+		f111 = buffer.Read<uint8_t>(1);
+
+		if (f111)
+		{
+			f100 = buffer.Read<int>(8) / 255.0f;
+		}
+		else
+		{
+			f100 = -1;
```

Comment:
`f100` is a float, if `f111` statement is false, `f100` is being set to ` 0xBF800000`, which is `-1.0f`.